v2/internal/genericarmclient/policy_register_rp.go

```diff
@@ -0,0 +1,221 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package genericarmclient
+
+// NOTE: This file was copied from https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/arm/runtime/policy_register_rp.go
+// with modifications. The primary modification was to remove all retries and immediately report an error after registration.
+// That error will be categorized and retried on by the operator.
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"reflect"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	armpolicy "github.com/Azure/azure-sdk-for-go/sdk/azcore/arm/policy"
+	armruntime "github.com/Azure/azure-sdk-for-go/sdk/azcore/arm/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud"
+	azpolicy "github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+
+	"github.com/Azure/azure-service-operator/v2/internal/version"
+)
+
+// NewRPRegistrationPolicy creates a policy object configured using the specified options.
+// The policy controls whether an unregistered resource provider should automatically be
+// registered. See https://aka.ms/rps-not-found for more information.
+func NewRPRegistrationPolicy(cred azcore.TokenCredential, o *azpolicy.ClientOptions) (azpolicy.Policy, error) {
+	conf, err := getConfiguration(o)
+	if err != nil {
+		return nil, err
+	}
+	authPolicy := armruntime.NewBearerTokenPolicy(cred, &armpolicy.BearerTokenOptions{Scopes: []string{conf.Audience + "/.default"}})
+	p := &rpRegistrationPolicy{
+		endpoint: conf.Endpoint,
+		pipeline: runtime.NewPipeline("generic", version.BuildVersion, runtime.PipelineOptions{PerRetry: []azpolicy.Policy{authPolicy}}, o),
+		options:  o,
+	}
+	return p, nil
+}
+
+type rpRegistrationPolicy struct {
+	endpoint string
+	pipeline runtime.Pipeline
+	options  *azpolicy.ClientOptions
+}
+
+func (r *rpRegistrationPolicy) Do(req *azpolicy.Request) (*http.Response, error) {
+	const unregisteredRPCode = "MissingSubscriptionRegistration"
```

Comment:
Suggest adding a comment to indicate what this constant is for
```suggestion
        // The unique identifier returned by ARM when the required resource provider has not been registered
	const unregisteredRPCode = "MissingSubscriptionRegistration"
```